net/memnet/memnet.go

```diff
@@ -6,3 +6,84 @@
 // in tests and other situations where you don't want to use the
 // network.
 package memnet
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/netip"
+	"sync"
+
+	"tailscale.com/net/netx"
+)
+
+// Net returns a [netx.Network] implementation that uses an in-memory
+// network for testing. It is only suitable for tests that do not
+// require real network access.
+//
+// As of 2025-04-08, it only supports TCP.
+func New() netx.Network {
+	return &memNetwork{}
+}
+
+// memNetwork implements [Network] using an in-memory network.
+type memNetwork struct {
```

Comment:
No deep objection, but is there some reason we can't export and return the concrete type here? (I realize this is mostly just code movement, I just noticed it in passing)

Reply:
We probably could now... it has a nice name (memnet.Network).

And that'd permit adding memnet.Network-specific methods.

Will do.